### PR TITLE
Dynamic Links - updates/tests

### PR DESCRIPTION
**Pull request by Ehesp**

### CODE REVIEW

File: packages/dynamic-links/lib/index.d.ts [hunk 1]
```diff
@@ -135,63 +124,59 @@ export namespace DynamicLinks {
...
      */
-    setAppStoreId(appStoreId: string): DynamicLinkParameters;
+    appStoreId?: string;
 
     /**
-     * Sets the iOS bundle ID.
+     * The bundle ID of the iOS app to use to open the link. The app must be connected to your project from the Overview page of the Firebase console.
      *
-     * @param bundleId The bundle ID of the iOS app to use to open the link. The app must be connected to your project from the Overview page of the Firebase console.
+     * This option is required if passing android options to your dynamic link.
```
Comment (Salakar) [inline packages/dynamic-links/lib/index.d.ts]:
```suggestion
     * This option is required if passing iOS options to your dynamic link.
```

File: packages/dynamic-links/lib/index.js [hunk 2]
```diff
@@ -55,9 +49,24 @@ class FirebaseLinksModule extends FirebaseModule {
...
+      shortLinkType !== 'DEFAULT' ||
+      shortLinkType !== 'SHORT' ||
+      shortLinkType !== 'UNGUESSABLE'
+    ) {
+      throw new Error(
+        `firebase.dynamicLinks().buildShortLink(_, *) 'shortLinkType' expected one of DEFAULT, SHORT or UNGUESSABLE.`,
+      );
+    }
+
+    return this.native.buildLink(params, shortLinkType);
```
Comment (Salakar) [inline packages/dynamic-links/lib/index.js]:
```suggestion
    return this.native.buildShortLink(params, shortLinkType);
```

File: packages/dynamic-links/lib/index.js [hunk 3]
```diff
@@ -21,31 +21,25 @@ import {
   getFirebaseRoot,
 } from '@react-native-firebase/app/lib/internal';
 
-import DynamicLinkParameters from './DynamicLinkParameters';
-
+import builder from './builder';
 import version from './version';
 
-const statics = {
```
Comment (Salakar) [inline packages/dynamic-links/lib/index.js]: Removing statics will mean statics is undefined on line 100, need to keep this and just empty object it.

File: packages/dynamic-links/lib/index.js (hunk 2 above)
Comment (Salakar) [inline packages/dynamic-links/lib/index.js]:
```suggestion
    return this.native.buildShortLink(params, shortLinkType);
```

**Comment by Ehesp** [comment]
cc @Salakar updated with changes

**Pull request merged**